colossalai/fx/passes/shard_1d_pass.py

```diff
@@ -26,28 +28,31 @@ def all_reduce_function(input_):
     return input_
 
 
-def weight_split(weight, dim):
-    #TODO: this function will be refactored by using ColoTensor dist_spec when a stable reshaper feature is ready to use.
-    num_partition = gpc.get_world_size(ParallelMode.TENSOR)
-    shape = weight.shape
-    length = shape[dim] // num_partition
-    sharded_weight_list = []
-    for i in range(num_partition):
-        sharded_weight_list.append(weight.narrow(dim, i * length, length))
-    return sharded_weight_list[gpc.get_local_rank(ParallelMode.PARALLEL_1D)]
+def weight_split(weight):
+    # split tensor with tensor_spec
+    assert isinstance(weight, colossalai.tensor.colo_tensor.ColoTensor), \
+        f'The type of the input tensor should be colotensor' \
+        f'Your Input tensor is {type(weight)}'
+    # assert weight.device.type == "meta"
+    tensor_dist_spec = weight.tensor_spec
+    num_partition = tensor_dist_spec.get_process_group_size()
```

Comment:
The API has changed!
current it should be
```
tensor_dist_spec.process_group.tp_world_size()
```

Reply:
Thank you. I will check that!